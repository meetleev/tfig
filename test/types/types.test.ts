
import ps from 'path';
import fs from 'fs-extra';
import { bundle } from '../../source/gift';

test('From triple slash directive', async () => {
    const inputPath = ps.join(__dirname, 'input.d.ts');
    const outputPath = ps.join(__dirname, 'output.d.ts');
    const { groups } = bundle({
        input: inputPath,
        rootModule: 'index',
        name: 'cc',
        output: outputPath,
    });
    expect(groups.length).toBe(1);
    const group0 = groups[0];

    expect(group0.code).toMatchSnapshot();
});