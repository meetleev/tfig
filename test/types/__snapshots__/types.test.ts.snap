// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`From triple slash directive 1`] = `
"declare module \\"cc\\" {
    export class Component {
    }
    export class Node {
        getComponent<T extends Component>(constructor: __private._types_globals__Constructor<T>): T | null;
    }
    export namespace __private {
        export type _types_globals__Constructor<T = unknown> = new (...args: any[]) => T;
    }
    export {};
}
"
`;
